### ISSUE TITLE
[debugger and emulator crash] Surely by misuse but severe crash in any case

### ISSUE THREAD

**Issue by Bunkai9448** [mgba-emu/mgba#2900]
The debugger lets you input negative values for the trace, which cause unexpected behavior of the emulator. Always resulting in the emulator overloading, a c++ runtime error and emulator crash.

I know I shouldn't have tried that by I didn't know if there was a way to see the instruction prior to a reached watch point, and I decided to put that to see what happened out of curiosity.

Steps to reproduce:
(With a game loaded)

Go to Tools > Open debugger console…
type `trace -1`

(Again, sorry for trying this, I know this doesn't make sense for someone with a minimum knowledge of how this work. I expected a "command not valid" or some kind of try-catch with info on the stack of the emulator).

PS: I leave you a pdf of what I was doing before that if it helps in any way for
future use.
[mgba pointers.pdf](https://github.com/mgba-emu/mgba/files/11327492/mgba.pointers.pdf)

**Comment by endrift** [comment]
It doesn't crash on my end, it just gets really slow. Plus, if you hit the `break` button it'll eventually respond, then you can cancel the trace by typing `trace 0`.

-1 might be getting converted into MAX_UINT, roughly 4 billion. I'd have to check. While negative values don't make sense, there's no real "sensible" upper bound here, so I'm not entirely sure how to handle this.

**Comment by Bunkai9448** [comment]
I see, thanks for the answer.

I'll add a bit of info in case it helps.

(Knowing the way you handle the trace this is mostly redundant, but I did before reading this, so I'll put it as extra) I tried a few more negatives values with the same result, again just as curiosity. 
![IMG_20230426_095129](https://user-images.githubusercontent.com/71356234/234507300-dba30709-cb43-497d-8ab8-2946124a3cf3.jpg)

PS: [I hope this doesn't derail much, but I have a somwhow related question]
Since I am going to do it the proper way, and I was thinking into using the proper trace use in the alternative (reseting and doing the trace prior those instructions/cicles).

I took into account using ghidra, but discarded as the first option due to the overwork, (will do if it's the only option tho)

My idea with the current situation is:
Reset, pause before the watchpoint moment and trace. (it requires an input of "A button" in between" if it makes a difference).

And I was considering writing a mini-script to save the info. (I didn't see a specific function, so I'll ask) is there any way to do the trace in the scripts or do I have to go by manually doing it instead??
